ui-lib/src/main/java/co/electriccoin/zcash/ui/common/Button.kt

```diff
@@ -0,0 +1,56 @@
+package co.electriccoin.zcash.ui.common
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.PressInteraction
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.material3.Button
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+@Suppress("LongParameterList")
+@Composable
+fun TimedButton(
+    modifier: Modifier = Modifier,
+    duration: Duration = 5.seconds,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    coroutineDispatcher: CoroutineDispatcher = Dispatchers.Default,
```

Comment:
While I like default arguments for things that are test-only, I do have concerns when we are propagating them across many different layers like is happening in the recent changes.

It makes it much more likely that someone will pass in a different value at the top level but accidentally an intermediate layer forgets to pass that value through all the way to the end.

Can you provide some context why we're injecting a dispatcher and whether other options have been considered?

Reply:
1. Main reason is that I wasn't able to test TimedButton without passing the test dispatcher, because it launches a new coroutine to introduce the delay. I'm not sure if there's more elegant way to do this, but if you have something in mind please let me know.
2. As for default argument, i agree that it's not ideal to have default param on every layer, but are you saying that we should avoid using default params ?

Reply:
Did you try to run the delay above `Dispatchers.Default` and return it to `Dispatchers.Main` (as you already do)? We'd be able to get rid of the one from parameter.

Reply:
We discussed this question today and we came up with a few ideas. Basically, we have these ideas:
- Have the duration as a parameter and input there the real value (5s) for production calls and some test (e.g. 0.5s) while calling this api from test context
- We could also go with something similar to Android Views approach - measure a difference between press_down and press_up events in onTouchEvent callback and compare it with our constant value (5s)
- Or provide both callbacks in compose code. One for normal click and second long click and in the test SendViewTest we'd be ok with just click event, and in the separate TimedButtonTest we'd test the long press. It'd be useful, because I believe that we'll still need to catch also normal click to e.g. toast user that the long click is necessary or for showing some countdown to user, so he/she understands what time is needed.